packages/graphql/src/translate/where/create-node-where-and-params.ts

```diff
@@ -144,7 +148,76 @@ function createNodeWhereAndParams({
                 resultStr += recurse[0];
                 resultStr += ")"; // close NONE/ANY
                 res.clauses.push(resultStr);
-                res.params = { ...res.params, fieldName: recurse[1] };
+                res.params = { ...res.params, [fieldName]: recurse[1] };
+                return res;
+            }
+
+            if (connectionField) {
+                let nodeEntries: Record<string, any> = value;
+
+                if (!connectionField?.relationship.union) {
+                    nodeEntries = { [connectionField.relationship.typeMeta.name]: value };
+                }
+
+                Object.entries(nodeEntries).forEach((entry) => {
```

Comment:
**suggestion (non-blocking):** This callback is a prime candidate to be extracted into a (fully) separate function to reduce the complexity of the `reduce` function (we are easily surpassing a cyclomatic complexity of 5 here), size, and scope of its variables.